[state store ttl] Add context to persisting state (#3546)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/state-management/state-store-ttl.md

When a TTL is not specified, the default behavior of the state store is retained.

## Explicit persistence bypassing globally defined TTL

Persisting state applies to all state stores that let you specify a default TTL used for all data, either:

> - At the Dapr component level, or 

Reviewer: ```suggestion
- Setting a global TTL value via a Dapr component, or 
```

> - When creating the resource outside of Dapr. 

Reviewer: ```suggestion
- When creating the state store outside of Dapr and setting a global TTL value. 
```

> When no TTL is specified, the data expires after that default period of time. This is not facilitated by Dapr at all.

Reviewer: ```suggestion
When no specific TTL is specified, the data expires after that global TTL period of time. This is not facilitated by Dapr.
```

> In addition, all of these state stores also support the option to _explicitly_ persist data. This means they ignore the default database policy (which may or may not have been set up using Dapr) to indefinitely retain a given database record. You can do this by setting `ttlInSeconds` to the value `-1`.

Reviewer: ```suggestion
In addition, all state stores also support the option to _explicitly_ persist data. This means you can ignore the default database policy (which may have been set outside of Dapr or via a Dapr Component) to indefinitely retain a given database record. You can do this by setting `ttlInSeconds` to the value of `-1`. This value indicates to ignore any TTL value set.
```
